art/classifiers/tensorflow.py

```diff
@@ -178,6 +178,10 @@ def fit(self, x, y, batch_size=128, nb_epochs=10, **kwargs):
         # Apply preprocessing
         x_preprocessed, y_preprocessed = self._apply_preprocessing(x, y, fit=True)
 
+        # Check label shape
+        if len(self._labels_ph.shape) == 1:
```

Comment:
How about moving this condition to __init__ and set with it attribute `self._reduce_labels` to make it consistent across classifiers?

Reply:
Done.